From 78e1c48805e0087a1103d975b7078abdf6eea1db Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Fri, 15 Dec 2023 10:08:33 +0530
Subject: [PATCH] Fixed for Crash in KitchenSink app During Monkey Test.

During monkey test run, observing following crash
android.util.AndroidRuntimeException: Calling startActivity() from
outside of an Activity context requires the FLAG_ACTIVITY_NEW_TASK flag.
Is this really what you want?
at android.app.ContextImpl.startActivity(ContextImpl.java:1088)
at android.app.ContextImpl.startActivity(ContextImpl.java:1064)
at android.content.ContextWrapper.startActivity(ContextWrapper.java:411)
at com.google.android.car.kitchensink.UserNoiticeDemoUiService
.startDialog(UserNoiticeDemoUiService.java:122)

we cannot start an activity from a non-activity context unless you
pass the intent flag FLAG_ACTIVITY_NEW_TASK. If you attempt to start an
activity without passing this flag.

Tests:
Run monkey test and observe, test is pass.

Tracked-On: OAM-113423
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 .../android/car/kitchensink/UserNoiticeDemoUiService.java      | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/EmbeddedKitchenSinkApp/src/com/google/android/car/kitchensink/UserNoiticeDemoUiService.java b/tests/EmbeddedKitchenSinkApp/src/com/google/android/car/kitchensink/UserNoiticeDemoUiService.java
index 502c9bca9c..47c9147181 100644
--- a/tests/EmbeddedKitchenSinkApp/src/com/google/android/car/kitchensink/UserNoiticeDemoUiService.java
+++ b/tests/EmbeddedKitchenSinkApp/src/com/google/android/car/kitchensink/UserNoiticeDemoUiService.java
@@ -119,7 +119,8 @@ public class UserNoiticeDemoUiService extends Service {
             if (!Settings.canDrawOverlays(this)) {
                 Intent intent = new Intent();
                 intent.setAction(Settings.ACTION_MANAGE_OVERLAY_PERMISSION);
-                startActivity(intent);
+                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                getApplicationContext().startActivity(intent);
                 return;
             }
             mDialog = createDialog();
-- 
2.17.1

